packages/state/core/README.md

```diff
@@ -0,0 +1,55 @@
+# @palantir/pack.state.core
+
+Real-time state management system.
+
+## Overview
+
+This package provides the core state management infrastructure for PACK, built around the concept of Documents that can be synchronized in real-time across multiple clients.
+
+`StateModule` via `app.state` is the main public interface.
+
+## Key Concepts
+
+### Terminology
+
+> TODO: These need another pass for consistent naming, there's a confusing interchange of Schema & Model between document & record layer.
```

Comment:
@alexdlm thoughts on if we should do any of this now? Do we you have a crisp idea of what your ideal terminology is?